### ISSUE TITLE
Fix a bug when there are only DIA-Quant data and the spectral library file, FragPipe throws an exception.

### ISSUE THREAD

**Issue by Arthfael** [Nesvilab/FragPipe#1334]
I am trying to setup FragPipe searches on our new dedicated Linux node. We have installed and configured FragPipe as a module on our HPC and can start the GUI from the Linux desktop by opening a terminal and running:
```
> module load fragpipe
> fragpipe
```
The GUI opens up - and I can confirm that in the config tab all tools are correctly configured. I then load the DIA_SpecLib_Quant workflow, load my files, assign them DIA-Quant, load a fasta and a .speclib... when I hit Run, I get reproducibly the following message in the terminal (not a popup):
"There are no DDA, DIA, GPF-DIA, or GPF-Lib" and the run does not start.
I am surprised by this, as normally with a library for DIA-Quant files I should not need the other file types. The same works for us on Windows.

**Comment by fcyu** [comment]
Thanks for the feedback. Could you share the log file? Did you have other module enabled? In an old version of FragPipe, we do require non-DIA-Quant files, but I think I have changed that. Need to see which version are you using.

Best,

Fengchao

**Comment by Arthfael** [comment]
Dear Fengchao,

Thank you very much for the swift answer.

This is FragPipe 20.0, running on Debian.

Unfortunately it does not appear that FragPipe went far enough to actually save a log. The GUI does not crash, but after the usual checks (whether the output directory should be created or, if existing, whether it should be cleaned of existing files) the run auto-aborts. I can see sub-directories for each Experiment in the folder but they are empty too.

I did not load any other modules, but I guess some must be auto-loaded. Here is the output of lsmod:
```
...@...:~$ lsmod
Module                  Size  Used by
tcp_diag               16384  0
inet_diag              24576  1 tcp_diag
unix_diag              16384  0
binfmt_misc            24576  1
dm_mod                163840  0
nfsv3                  57344  13
nfs_acl                16384  1 nfsv3
nfs                   434176  99 nfsv3
lockd                 126976  2 nfsv3,nfs
grace                  16384  1 lockd
nfs_ssc                16384  1 nfs
fscache               397312  1 nfs
8021q                  40960  0
garp                   16384  1 8021q
stp                    16384  1 garp
mrp                    20480  1 8021q
llc                    16384  2 stp,garp
amd64_edac_mod         40960  0
edac_mce_amd           32768  1 amd64_edac_mod
kvm_amd               118784  0
rfkill                 32768  1
ipmi_ssif              40960  0
kvm                   937984  1 kvm_amd
irqbypass              16384  1 kvm
ghash_clmulni_intel    16384  0
aesni_intel           372736  0
joydev                 28672  0
libaes                 16384  1 aesni_intel
crypto_simd            16384  1 aesni_intel
cryptd                 24576  2 crypto_simd,ghash_clmulni_intel
glue_helper            16384  1 aesni_intel
hid_generic            16384  0
wmi_bmof               16384  0
rapl                   20480  0
efi_pstore             16384  0
pcspkr                 16384  0
nls_ascii              16384  1
nls_cp437              20480  1
vfat                   20480  1
fat                    86016  1 vfat
usbhid                 65536  0
hid                   151552  2 usbhid,hid_generic
acpi_ipmi              20480  0
ccp                   110592  1 kvm_amd
sp5100_tco             20480  0
rng_core               16384  1 ccp
k10temp                16384  0
watchdog               32768  1 sp5100_tco
ipmi_si                73728  1
ipmi_devintf           20480  0
ipmi_msghandler       118784  4 ipmi_devintf,ipmi_si,acpi_ipmi,ipmi_ssif
acpi_cpufreq           32768  0
evdev                  28672  7
nf_log_ipv6            16384  5
ip6t_REJECT            16384  1
nf_reject_ipv6         20480  1 ip6t_REJECT
xt_hl                  16384  22
ip6_tables             36864  52
ip6t_rt                20480  3
nf_log_ipv4            16384  5
nf_log_common          16384  2 nf_log_ipv4,nf_log_ipv6
ipt_REJECT             16384  1
nf_reject_ipv4         16384  1 ipt_REJECT
xt_LOG                 20480  10
nft_limit              16384  13
ib_mthca              163840  0
rdma_ucm               32768  0
rpcrdma               323584  26
rdma_cm               131072  2 rpcrdma,rdma_ucm
iw_cm                  57344  1 rdma_cm
ib_umad                36864  0
xt_limit               16384  0
xt_addrtype            16384  4
xt_tcpudp              20480  24
xt_conntrack           16384  16
nf_conntrack          176128  1 xt_conntrack
nf_defrag_ipv6         24576  1 nf_conntrack
nf_defrag_ipv4         16384  1 nf_conntrack
nft_compat             20480  141
nft_counter            16384  176
ib_ipoib              147456  0
ib_cm                 135168  2 rdma_cm,ib_ipoib
nf_tables             266240  407 nft_compat,nft_counter,nft_limit
mlx4_ib               225280  0
libcrc32c              16384  2 nf_conntrack,nf_tables
mlx4_core             380928  1 mlx4_ib
nfnetlink              20480  2 nft_compat,nf_tables
parport_pc             40960  0
ppdev                  24576  0
lp                     20480  0
parport                73728  3 parport_pc,lp,ppdev
fuse                  167936  3
configfs               57344  2 rdma_cm
sunrpc                667648  248 rpcrdma,lockd,nfsv3,nfs_acl,nfs
efivarfs               16384  1
ip_tables              36864  8
x_tables               53248  12 xt_conntrack,nft_compat,xt_LOG,xt_tcpudp,xt_addrtype,ip6t_rt,ip6_tables,ipt_REJECT,ip_tables,xt_limit,xt_hl,ip6t_REJECT
autofs4                53248  2
ext4                  937984  4
crc16                  16384  1 ext4
mbcache                16384  1 ext4
jbd2                  151552  1 ext4
crc32c_generic         16384  0
ast                    57344  1
drm_vram_helper        24576  1 ast
mlx5_ib               385024  0
drm_ttm_helper         16384  1 drm_vram_helper
i2c_algo_bit           16384  1 ast
ib_uverbs             167936  4 mlx4_ib,ib_mthca,rdma_ucm,mlx5_ib
ttm                   114688  2 drm_vram_helper,drm_ttm_helper
ib_core               413696  11 rdma_cm,ib_ipoib,rpcrdma,mlx4_ib,iw_cm,ib_mthca,ib_umad,rdma_ucm,ib_uverbs,mlx5_ib,ib_cm
mlx5_core            1183744  1 mlx5_ib
drm_kms_helper        278528  4 drm_vram_helper,ast
nvme                   49152  6
ahci                   40960  0
xhci_pci               24576  0
nvme_core             131072  8 nvme
libahci                45056  1 ahci
cec                    61440  1 drm_kms_helper
mlxfw                  32768  1 mlx5_core
xhci_hcd              311296  1 xhci_pci
crc32_pclmul           16384  0
t10_pi                 16384  1 nvme_core
ptp                    32768  1 mlx5_core
crc_t10dif             20480  1 t10_pi
crc32c_intel           24576  9
libata                299008  2 libahci,ahci
crct10dif_generic      16384  0
usbcore               331776  3 xhci_hcd,usbhid,xhci_pci
crct10dif_pclmul       16384  1
pps_core               24576  1 ptp
drm                   630784  7 drm_kms_helper,drm_vram_helper,ast,drm_ttm_helper,ttm
bnxt_en               294912  0
scsi_mod              270336  1 libata
crct10dif_common       16384  3 crct10dif_generic,crc_t10dif,crct10dif_pclmul
pci_hyperv_intf        16384  1 mlx5_core
i2c_piix4              28672  0
usb_common             16384  2 xhci_hcd,usbcore
wmi                    36864  1 wmi_bmof
button                 24576  0
```


Best regards,

Armel

**Comment by Arthfael** [comment]
I forgot to say, but I did check just now: I did tick IM-MS in the Workflow tab.

**Comment by fcyu** [comment]
Hi Armel,

Could you check if you have MSBooster enabled? If you are not sure, could you share your workflow file?

Thanks,

Fengchao

**Comment by Arthfael** [comment]
Dear Fengchao,

I think it is, here is a relevant extract workflow from a recent successful DDA IM-MS run:
```
# FragPipe (20.0) runtime properties
...
msbooster.predict-rt=true
msbooster.predict-spectra=true
msbooster.run-msbooster=true
msbooster.use-correlated-features=false
...

```
Kind regards,

Armel

**Comment by fcyu** [comment]
Hi Armel,

Yeah, you enabled MSBooster, which you should not and it caused your problem. Can you set `msbooster.run-msbooster=false` and try again?

You probably also need to check other parameters. The most straightforward way is loading the workflow file in FragPipe GUI and inspect the settings.

Best,

Fengchao

**Comment by Arthfael** [comment]
Dear Fengchao,

It seems MSBooster was activated by default when I loaded the default workflow - I did not touch the Validation tab.
I have now de-activated it. This lets me progress to the next error ^^ Now, I get a java error popup:
```
java.util.NoSuchElementException
at java.base/java.util.HashMap$HashIterator.nextNode(HashMap.java:1513)
at java.base/java.util.HashMap$KeyIterator.next(HashMap.java:1534)
at com.dmtavt.fragpipe.cmd.CmdProteinProphet.configure(CmdProteinProphet.java:144)
at com.dmtavt.fragpipe.FragpipeRun.lambda$configureTaskGraph$37(FragpipeRun.java:1318)
at com.dmtavt.fragpipe.FragpipeRun.configureTaskGraph(FragpipeRun.java:1801)
at com.dmtavt.fragpipe.FragpipeRun.run(FragpipeRun.java:294)
at com.dmtavt.fragpipe.tabs.TabRun.on(TabRun.java:178)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:517)
at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:511)
at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:829)
```

In terminal, the error is the same but preceded by:
`[SEVERE] Could not dispatch event: class com.dmtavt.fragpipe.messages.MessageRun to subscribing class class com.dmtavt.fragpipe.tabs.TabRun`

This seems related to [https://github.com/Nesvilab/FragPipe/issues/235](url)?

I have attached my workflow (MSBooster now off; somehow I had to change the extension to be able to upload it to github, otherwise it was not showing in the upload popup: strange...)

Best,

Armel



On Thu, Nov 16, 2023 at 4:27 PM Fengchao <notifications@github.com> wrote:

    Hi Armel,

    Yeah, you enabled MSBooster, which you should not and it caused your problem. Can you set msbooster.run-msbooster=false and try again?

    You probably also need to check other parameters. The most straightforward way is loading the workflow file in FragPipe GUI and inspect the settings.

    Best,

    Fengchao

**Comment by fcyu** [comment]
Hi Armel,

Some settings in your workflow broke FragPipe.

I couldn't find your workflow. You need to zip it and upload it to https://github.com/Nesvilab/FragPipe/issues/1334

Best,

Fengchao

**Comment by Arthfael** [comment]
[DIA_test.zip](https://github.com/Nesvilab/FragPipe/files/13379832/DIA_test.zip)
Here you go ^^

**Comment by fcyu** [comment]
Thanks for the file. For the current version, if only having the `DIA-Quant` data and provide the spectral library file, you need to disable all modules, such as MSFragger, Validation, Quant (MS1), except for the Quant (DIA). I have modified your workflow and put it here [DIA_test_yufe.zip](https://github.com/Nesvilab/FragPipe/files/13380028/DIA_test_yufe.zip)

BTW, it seems that the fasta file and the spectral library file have different path format: `G:\nfs\scistore13\lsfgrp\anicolas\Test_FragPipe_on_Linux\2023-11-15-decoys-contam-Homo_sapiens_UP_000005640_20230228_iso.fasta.fas1` VS `/nfs/scistore13/lsfgrp/anicolas/Test_DiaNN/First_success/wholeproteome_crap_predlib.tsv.speclib`. Not sure if it matters. Need to double check.

Best,

Fengchao

**Comment by Arthfael** [comment]
Dear Fengchao,

Thank you so much. Since the DIA_SpecLib_Quant did not disable those tabs
by default, I had not assumed that this was necessary. Once I disable them,
I am able to run normally.

Strange about the file format. I did reload the Workflow from a Workflow
which had been saved on a Windows machine (which does not have any drive
mapped to G: though), but then in Linux I browsed to the correct locations
for all inputs from without the GUI, including the fasta. I did not
manually edit paths in the Workflow.

As usual, you have been immensely helpful, thank you so much!

Kind regards,

Armel
